Ovmf acrn gop

xinyunliu opened the pull request.

xinyunliu committed:
OvmfPkg/GvtGopDxe: Add GvtGopDxe skeleton driver

This driver provides a UEFI Graphics Output Protocol (GOP) driver for
Intel GVT-g VGA hardware. It simulates a 24-bit color framebuffer, and
uses the standard 32-bit GOP pixel format whenever possible.

Tracked-On: projectacrn/acrn-hypervisor#3624
Co-Developed-by: He Min <min.he@intel.com>
Signed-off-by: Liu Xinyun <xinyun.liu@intel.com>

peterfang commented:
Looks good overall. 

Are there ACRN-specific values/implementations in this GVT driver? If there are, I'd prefer that we separate them out or at least put an "Acrn" prefix in them.

The only other feedback I have is to conform with the EDK II coding style.
More specifically:

1. Use CamelCase for your struct and variables
2. Always put space before an open parenthesis (please refer to 5.2.2.6 in the coding style doc)

For more info: https://github.com/tianocore/tianocore.github.io/wiki/Code-Style-C

Thanks!

xinyunliu commented:
Hi Peter,
Thanks for the review.

1. This implementation can be reused without modification for kvmgt/xengt if we port the linux kernel changes from acrngt. 
2.  There are some structure taken from linux kernel, so does not follow OVMF coding style, should them be transfered to OVMF style?
'''
typedef struct {
  UINT64 magic;                 /* VGT_MAGIC */
  UINT16 version_major;
  UINT16 version_minor;
  UINT32 vgt_id;
  UINT32 vgt_caps;              /* VGT capabilities */
} GVT_IF_HDR;

typedef struct {
  UINT32        fb_base;
  UINT32        width;
  UINT32        height;
  UINT32        pitch;
  UINT32        bpp;
  UINT32        size;
} GVT_GOP_INFO;
'''

3. Will fix the missing pace before an open parenthesis.

> Looks good overall.
> 
> Are there ACRN-specific values/implementations in this GVT driver? If there are, I'd prefer that we separate them out or at least put an "Acrn" prefix in them.

> 
> The only other feedback I have is to conform with the EDK II coding style.
> More specifically:
> 
> 1. Use CamelCase for your struct and variables
> 2. Always put space before an open parenthesis (please refer to 5.2.2.6 in the coding style doc)
> 
> For more info: https://github.com/tianocore/tianocore.github.io/wiki/Code-Style-C
> 
> Thanks!

peterfang commented:
> Hi Peter,
> Thanks for the review.
> 
> 1. This implementation can be reused without modification for kvmgt/xengt if we port the linux kernel changes from acrngt.
That's great!

> 2. There are some structure taken from linux kernel, so does not follow OVMF coding style, should them be transfered to OVMF style?
Yes, please convert them to OVMF's coding style so they stay consistent.
Perhaps you can add a comment there so we know they come from the Linux kernel.

>    '''
>    typedef struct {
>    UINT64 magic;                 /* VGT_MAGIC _/
>    UINT16 version_major;
>    UINT16 version_minor;
>    UINT32 vgt_id;
>    UINT32 vgt_caps;              /_ VGT capabilities */
>    } GVT_IF_HDR;
> 
> typedef struct {
> UINT32 fb_base;
> UINT32 width;
> UINT32 height;
> UINT32 pitch;
> UINT32 bpp;
> UINT32 size;
> } GVT_GOP_INFO;
> '''
> 
> 1. Will fix the missing pace before an open parenthesis.
Thanks!

> 
> > Looks good overall.
> > Are there ACRN-specific values/implementations in this GVT driver? If there are, I'd prefer that we separate them out or at least put an "Acrn" prefix in them.
> 
> > The only other feedback I have is to conform with the EDK II coding style.
> > More specifically:
> > 
> > 1. Use CamelCase for your struct and variables
> > 2. Always put space before an open parenthesis (please refer to 5.2.2.6 in the coding style doc)
> > 
> > For more info: https://github.com/tianocore/tianocore.github.io/wiki/Code-Style-C
> > Thanks!

xinyunliu committed:
GvtGopDxe: Enable GOP with parameters from vgt interface

Notify GVT kernel driver to set up framebuffer for GOP and get mode
information from vgt interface

Tracked-On: projectacrn/acrn-hypervisor#3624
Co-Developed-by: He Min <min.he@intel.com>
Signed-off-by: Liu Xinyun <xinyun.liu@intel.com>
